src/herder/Herder.h

```diff
@@ -67,8 +67,14 @@ class Herder
 
     enum State
     {
+        // Starting up, no state is known
+        HERDER_BOOTING_STATE,
+        // Fell out of sync, resyncing
         HERDER_SYNCING_STATE,
-        HERDER_TRACKING_STATE,
+        // Never been in sync with the network, track LCL
+        HERDER_TRACKING_LCL_STATE,
```

Comment:
should `HERDER_TRACKING_LCL_STATE` be defined before `HERDER_SYNCING_STATE` ?

I imagine that later on, the only valid transition is to bounce between `HERDER_TRACKING_LCL_STATE` and `HERDER_TRACKING_NETWORK_STATE`

Reply:
Also, I find it confusing that `HERDER_TRACKING_LCL_STATE` is called "tracking" - I'd prefer to call it `HERDER_BOOTING_LCL_STATE` as it should only be in that state until it sees consensus.
That way the node can then bounce between `HERDER_SYNCING_STATE` and `HERDER_TRACKING_NETWORK_STATE`

Reply:
> Also, I find it confusing that HERDER_TRACKING_LCL_STATE is called "tracking" - I'd prefer to call it HERDER_BOOTING_LCL_STATE as it should only be in that state until it sees consensus.
> That way the node can then bounce between HERDER_SYNCING_STATE and HERDER_TRACKING_NETWORK_STATE

This one is less clear to me. From the node's perspective, "tracking" means being in sync, i.e. ready to validate slots and participate in consensus. It seems cleaner to me to keep calling the state restored from DB as "tracking", not "booting", because as soon as node reaches that state, it's ready to vote (and it doesn't necessarily need to hear from the network to do so). This naming conversion also improves clarity and readability of helper functions such as `isTracking` (if we rename this state, this function becomes "HERDER_TRACKING_NETWORK_STATE || HERDER_BOOTING_LCL_STATE", which is pretty confusing). 

As I'm writing this, I'm wondering - maybe the issue is that both states should indicate that the node's Herder is synced? Maybe the state names should be something like "SYNCED_LOCAL_LCL" and "SYNCED_TRACKING_NETWORK", and `isTracking` can be renamed to `isSynced`. what do you think?